File: leetcode_vs/9.回文数.py
#
# @lc app=leetcode.cn id=9 lang=python3
#
# [9] 回文数
#

# @lc code=start


class Solution:
    def isPalindrome(self, x: int) -> bool:
        num = []
        if x < 0:
            return False
        while x != 0:
            num.append(x % 10)
            x = x // 10
        left, right = 0, len(num) - 1
        print(num)
        while left < right:
            if num[left] != num[right]:
                return False
            left += 1
            right -= 1
        return True


# @lc code=end
if __name__ == "__main__":
    solution = Solution()
    print(solution.isPalindrome(121))

File: 9.回文数.java
/*
 * @lc app=leetcode.cn id=9 lang=java
 *
 * [9] 回文数
 */

// @lc code=start
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0)
            return false;
        int tmp = x;
        int res = 0;
        while (tmp > 0) {
            res = res * 10 + tmp % 10;
            tmp /= 10;
        }
        return res == x;
    }
}
// @lc code=end

